api.php

```diff
@@ -154,36 +176,25 @@
 	### ### ####
 	*/
 
-	//regex pattern set to a string
-	$numberRegEx = '/\(?(\d{3})[\)\s-]*(\d{3})[\s\-]?(\d{4})/';
+
 	//check user input against string pattern
 	if (!preg_match($numberRegEx, $body->phone)) {
 		//set response message for invalid format
-		$response->getBody()->write(json_encode([
+		return badRequestResponse([
 			'field' => ['phone'],
 			'message' => 'invalid format for phone number',
-		]));
-		//return the error with an invalid request status
-		return $response
-			->withHeader('Content-Type', 'application/json')
-			->withStatus(400);
+		], $response);
 	}
-
 	//output of function set to userPhone w/ this format (###) ###-####
 	$userPhone = preg_replace($numberRegEx, '$1$2$3', $body->phone);
 	//if length of phone isn't equal to ten return an error
 	if (strlen($userPhone) !== 10) {
-		$response->getBody()->write(json_encode([
+		return badRequestResponse([
 			'field' => ['phone'],
 			'message' => 'phone number must be 10 digits'
-		]));
-		//return the error w/ status code
-		return $response
```

Comment:
I would like to go over in class why we need to check for the correct number format for phone, as well as length in two if statements. Code works tho

Reply:
this should be solved based on the work from yesterday